File: solutions/2025-dec/12/inventory_update.py
from collections import defaultdict


def update_inventory(inventory, shipment):
    items = defaultdict(int)
    for i in inventory:
        items[i[1]] += i[0]

    for s in shipment:
        items[s[1]] += s[0]

    ans = []
    for key, value in items.items():
        ans.append([value, key])

    return ans

File: solutions/2025-dec/12/inventory_update.js
function updateInventory(inventory, shipment) {
    const items = {};
    for (const i of inventory) {
        items[i[1]] ? (items[i[1]] += i[0]) : (items[i[1]] = i[0]);
    }
    for (const s of shipment) {
        items[s[1]] ? (items[s[1]] += s[0]) : (items[s[1]] = s[0]);
    }

    const ans = [];
    for (const [key, value] of Object.entries(items)) {
        ans.push([value, key]);
    }

    return ans;
}
